pkgs/top-level/appimage-packages.nix

```diff
@@ -0,0 +1,64 @@
+{ appimageTools, stdenv, fetchurl, fetchzip, fetchFromGitHub }:
+
+# fetchzip is for compatibility, bother AppImage builder about that :
+# https://docs.appimage.org/packaging-guide/distribution.html#do-not-put-appimages-into-other-archives
+
+
+let
+  self = appimagePackages;
+  appimagePackages = with self; {
+
+  buildAppImage = {
```

Comment:
I would prefer to just put packages in to `top-level/all-packages.nix` rather than one big expression file. After all packages should be built without using app images if possible. Having an abstraction is ok, but currently it does not seem to provide enough to justify the abstraction.

Reply:
> "all packages should be built without using app images if possible" 

we are very agree on that, but you know it's not always possible. 

> currently it does not seem to provide enough to justify the abstraction.

I made all my possible to put the interesting in appimageTools instead, but i think i get something useful here too, we still can someapp = appimagePackages.someapp to top-level/all-packages.nix , people would identify clearly it's not a native one.

Reply:
I think it makes most sense to merge it as is for now and introduce other features as needed, i've some on plan but not yet ready.

Reply:
Ok. Make it obvious that by doing `someapp = appimagePackages.someapp` looks like a good idea on the second thought - we should do this for all packages with no exception for visibility reasons though. However I would use callPackage in `appimage-packages.nix` for all in this for two reasons: It allows to override dependencies using `override` and one file per package is easier to maintain. We had at some point a huge `pkgs/top-level/python-packages.nix` which was not fun at all.

Reply:
yes the main interest was to simplify the maintainer and package work  . Perhaps i better move this buildappimage to appimagetools, and let people creating an one file per package, but the drawback is it's harder to maintain for me as author of buildAppImage. since i've some other nixpkgs job on the fire, i put this on wip again, let's this maturate.

Reply:
can't we have `buildAppImage` provided as a library function implementing all the somewhat repetitive bits of packaging a appimage as a nix derivation, and then still `callPackage` it from `all-packages.nix`?

I still like to be able to have packages in `all-packages.nix`, and be able to build them by building the top-level attribute.
Also, I like all the metadata we have, and that each package is in its own directory.

I still don't really see a point in a separate `appimage-packages.nix`, and having them inside a `appimagePackages.*` namespace - I'd say if there's some software only (reasonably) obtainable by extracting it from an appimage, then package it by extracting the appimage, but let's not make the way these are added into nixpkgs more alien than another regular package.

We already do that for some software, be it extracting a `.deb`, `snap` or whatever.

Reply:
Since we have ~150 recurseIntoAttrs in all-packages.nix, think in a first time that was possible to use that possibility for simplest appimages (not all). But a drawback is to not be able to use other callPackages like qt5.callPackage ( deprecated ? https://nixos.org/nix-dev/2017-February/022900.html ), libsForQt5 ... but i think that's not a real issue since this should make it unnecessary by the buildFHSUserEnv. ( what about python3Packages ? pkgsi686Linux ? ) . So i wonder if i have interest to provide a appimages.callPackage . So for the moment i will just move buildAppImage in appimageTools.